docs/decisions/0015-expiration-improvements.rst

```diff
@@ -0,0 +1,94 @@
+   the edge case where an assignment is allocated, cancelled, and re-allocated later.
+2. The current date is greater than the inferred enrollment deadline for the assigned course.
+3. The current date is greater than the expiration date of the subsidy associated with the Assignment's
+   access policy record.
+
+Learners are notified of expiration and cancellation via the learner-portal frontend
+------------------------------------------------------------------------------------
+The learner-portal notifies learners of assignment cancellation and expiration.  These notifications
+are required to be dismissable, which means we need to persist state about dismissal
+for assignments in the backend.
+
+Decision
+========
+
+``EXPIRED`` should be a distinct Assignment State
+-------------------------------------------------
+This will help disambiguate behavior and simplify/remove business logic
+from the learner-portal frontend.  We may wish to also store the expiration reason
+for an assignment (i.e. which of the 3 cases above caused the assignment to become expired).
+
+Persist the allocation, cancellation, and expiration dates
+----------------------------------------------------------
+This also helps simplify and clarify the true state of an assignment for
+consumption by the learner-portal frontend.
+
+Cancellation/Expiration acknowledgement should be an action
+-----------------------------------------------------------
+There should be an action type to support this, and an endpoint scoped to the assignment identifier
+to create such action records.  This will greatly simplify logic in the learner-portal frontend
+about the state of expiration/cancellation notifications and whether they have been acknowledged/dismissed
+by the learner.
+
+Serialize the expiration deadlines with assignments
+---------------------------------------------------
+Serialize the subsidy expiration date, enrollment deadline, and 90 day deadline in the serialized
+response of assignments inside the ``credits-available`` response payload (perhaps in the
+assignments list view response payload in the future).  Additionally, return a key like ``earliest-expiration``
+that is the minimum of these three dates.
+It's ok to not have a bulk API view to acknowledge multiple assignment expiration in a single request -
+these acknowledgements are one-time activities for most assignments, and we can live with the cost
+of making multiple requests from the frontend for now.  In return, we get to deal with a simpler API view
+and logic.
+
+Modify existing expiration command
+----------------------------------
+* Compare against the allocation date of assignments to determine 90 day expiration condition.
+* Set the state to ``EXPIRED`` and set the expiration date on the expired record.
+
+Consequences
+============
+* The difference between admin-directed cancellation vs. automatic expiration becomes obvious
```

Comment:
[note] This should actually get the behavior more in line with the intended UX 🎉